frontend/packages/dev-console/src/components/topology/componentUtils.ts

```diff
@@ -226,8 +242,27 @@ const edgeDragSourceSpec = (
 
 const createConnectorCallback = (serviceBinding: boolean) => (
   source: Node,
-  target: Node,
+  target: Node | Graph,
 ): any[] | null => {
+  const onKebabOptionClick = (option: KebabOption) => {
+    if (option.callback) {
+      option.callback();
+    }
+    if (option.href) {
+      const sourceObj = source.getData()?.resources?.obj || {};
+      const hrefWithContext = `${option.href}&contextSource=${referenceFor(sourceObj)}/${
+        sourceObj?.metadata?.name
+      }`;
+      history.push(hrefWithContext);
+    }
+  };
```

Comment:
We shouldn't have to override and duplicate behavior. Parameterize our actions like @karthikjeeyar did in `add-resources-menu-utils` to support `addApplication`. We should also support a context source object. This would put all the logic for creation of these actions in a single place.
The resulting code would be inside `#getAddPageUrl` to create the href.

I also do not think we need `isKnativeDisabled` as a query param in the URL. By @karthikjeeyar logic for the need, the fact that we'll have a `contextSource` will give us enough info to determine that we can hide the knative service option.

Reply:
updated the code to enable use of `add-resources-menu-utils`